packages/components/src/autocomplete/index.tsx

```diff
@@ -146,11 +90,20 @@ function useAutocomplete( {
 		if ( getOptionCompletion ) {
 			const completion = getOptionCompletion( option.value, filterValue );
 
-			const { action, value } =
-				undefined === completion.action ||
-				undefined === completion.value
-					? { action: 'insert-at-caret', value: completion }
-					: completion;
+			function isCompletionObject(
+				completionData: any
+			): completionData is OptionCompletion {
+				return ! (
+					undefined === completionData.action ||
+					undefined === completionData.value
+				);
```

Comment:
Just me, or is the negated `OR` harder to grok? 🙈

```suggestion
				return (
					undefined !== completionData.action &&
					undefined !== completionData.value
				);
```

Reply:
Agreed, thank you! 41ac316375